TigerConnectDemoApp/app/src/main/java/com/tigertext/ttandroid/sample/conversation/viewmodel/ConversationRepository.java

```diff
@@ -0,0 +1,48 @@
+package com.tigertext.ttandroid.sample.conversation.viewmodel;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+
+import com.tigertext.ttandroid.Message;
+import com.tigertext.ttandroid.RosterEntry;
+import com.tigertext.ttandroid.api.TT;
+import com.tigertext.ttandroid.http.GenericActionListener;
+import com.tigertext.ttandroid.org.Organization;
+
+import java.util.List;
+
+import timber.log.Timber;
+
+class ConversationRepository {
+
+    private MutableLiveData<List<Message>> messagesLiveData = new MutableLiveData<>();
+
+    MutableLiveData<List<Message>> getMessages() {
+        return messagesLiveData;
+    }
+
+    /**
+     * This is how you get a List of Messages to display in your conversation
+     * @param rosterEntry the Roster Entry of the conversation
+     * @param pageSize The number of messages you want for your conversation page
+     * @param topMessage The top message
+     */
+    void updateMessagesByPage(RosterEntry rosterEntry, int pageSize, Message topMessage) {
+        /**
+            This is how to get a specific conversation for a Roster Entry
+         */
+        TT.getInstance().getConversationManager().getMessagesByPage(rosterEntry, pageSize, topMessage, new GenericActionListener<List<Message>, Throwable>() {
+            @Override
+            public void onResult(List<Message> messages) {
+                // Post the messages in Live Data for observers to act on, or directly
+                // update the adapter/views with these new messages
+                messagesLiveData.postValue(messages);
```

Comment:
You don't have to change this, though I don't think this is a good use of LiveData. Observers of LiveData only get the updates when in the started or running state, and LiveData doesn't handle streams, it only gives the latest value to observers. In this case, it'll likely never be an issue because we only load new messages when the observer is running, and for demo purposes it gets the point across

Reply:
I see what you are saying. For messages this wouldn't be ideal since every update of the messages by page should be appending the new data on top of what is already inside the LiveData, not replacing it